src/linpred.jl

```diff
@@ -88,44 +84,32 @@ function delbeta!(p::DensePredQR{T,<:QRPivoted}, r::Vector{T}) where T<:BlasReal
     if rnk == length(p.delbeta)
         p.delbeta = p.qr\r
     else
-        R = @view p.qr.R[:, 1:rnk] 
-        Q = @view p.qr.Q[:, 1:size(R, 1)]
+        R = UpperTriangular(view(parent(p.qr.R), 1:rnk, 1:rnk))
         piv = p.qr.p
-        p.delbeta = zeros(size(p.delbeta))
-        p.delbeta[1:rnk] = R \ Q'r
+        fill!(p.delbeta, 0)
+        p.delbeta[1:rnk] = R \ view(p.qr.Q'r, 1:rnk)
         invpermute!(p.delbeta, piv)
     end
-    mul!(p.scratchm1, Diagonal(ones(size(r))), p.X)
     return p
 end
 
 function delbeta!(p::DensePredQR{T,<:QRPivoted}, r::Vector{T}, wt::Vector{T}) where T<:BlasReal
-    rnk = rank(p.qr.R)
     X = p.X
     W = Diagonal(wt)
     sqrtW = Diagonal(sqrt.(wt))
-    delbeta = p.delbeta
-    scratchm2 = similar(X, T)
     mul!(p.scratchm1, sqrtW, X)
-    mul!(scratchm2, W, X)
-    mul!(delbeta, transpose(scratchm2), r)
-
-    if rnk == length(p.delbeta)
-        qnr = qr(p.scratchm1)
-        Rinv = inv(qnr.R)
-        p.delbeta = Rinv * Rinv' * delbeta
-    else
-        qnr = pivoted_qr!(copy(p.scratchm1))
-        R = @view qnr.R[1:rnk, 1:rnk]
-        Rinv = inv(R)
-        piv = qnr.p
-        permute!(delbeta, piv)
-        for k=(rnk+1):length(delbeta)
-            delbeta[k] = -zero(T)
-        end
-        p.delbeta[1:rnk] = Rinv * Rinv' * view(delbeta, 1:rnk)
-        invpermute!(delbeta, piv)
+    r̃ = sqrtW * r
+
+    p.qr = pivoted_qr!(copy(p.scratchm1))
+    rnk = rank(p.qr.R) # FIXME! Don't use svd for this
+    R = UpperTriangular(view(parent(p.qr.R), 1:rnk, 1:rnk))
+    permute!(p.delbeta, p.qr.p)
+    for k = (rnk + 1):length(p.delbeta)
+        p.delbeta[k] = -zero(T)
     end
+    p.delbeta[1:rnk] = R \ (p.qr.Q'*r̃)[1:rnk]
```

Comment:
Maybe
```suggestion
    mul!(view(p.delbeta, 1:rnk), view(p.qr.Q, :, 1:rnk)', r̃)
    ldiv!(R, view(p.delbeta, 1:rnk))
```
?

Reply:
We can't make a view into `Q`, see above, but we can make a view into the result.